### ISSUE TITLE
Silent failure when uploading to Google Cloud Storage

### ISSUE THREAD

**Issue by kruckenb** [goreleaser/goreleaser#3043]
### What happened?

Uploads to Google Cloud Storage appeared to succeed, when they were actually failing due to missing permissions and a mis-named bucket in `.goreleaser.yml`

I expected that an error message would be output if the upload to GCS was not successful.

In both cases, the output from goreleaser was:

```
publishing       
      • blobs            
         • uploading                 path=upload/binary1
         • uploading                 path=upload/binary2
         • uploading                 path=upload/checksum.txt
```

But there was nothing in the bucket.

Once the bucket name was corrected 'Storage Object Creator' permissions was added, the files appeared in the bucket as expected.

### How can we reproduce this?

In .goreleaser.yml configured to upload blobs to Google Cloud Storage
- upload to a bucket that doesn't have 'Storage Object Creator' permissions
- upload to a bucket that doesn't exist

```yaml
blobs:
  - provider: gs
    bucket: non-existent-bucket
    extra_files:
      - glob: dist/binary1
      - glob: dist/binary2
    folder: "upload"
```

### goreleaser version

```bash
v1.7.0

FYI I can't easily run `goreleaser check` as this runs as an action in a CI/CD pipeline.
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
hey, thanks for the report... this was reported before and I couldn't reproduce it... since you have it a reproducible bucket, could you maybe work on a PR for it?

otherwise when I have some time I'll try to reproduce it again here

**Comment by caarlos0** [comment]
likely a dupe of https://github.com/goreleaser/goreleaser/issues/2652 

fix incoming in #3185
